compiler/code-gen/declarations.cpp

```diff
@@ -596,27 +637,167 @@ void ClassDeclaration::compile_get_hash(CodeGenerator &W, ClassPtr klass) {
   compile_class_method(FunctionSignatureGenerator(W).set_const_this(), klass, "int get_hash()", klass->get_hash());
 }
 
-void ClassDeclaration::compile_accept_visitor(CodeGenerator &W, ClassPtr klass, const char *visitor_type) {
+void ClassDeclaration::compile_accept_visitor(CodeGenerator &W, ClassPtr klass, std::string_view visitor_type) {
   compile_class_method(FunctionSignatureGenerator(W), klass, fmt_format("void accept({} &visitor)", visitor_type), "generic_accept(visitor)");
 }
 
+static KphpJsonTag::FieldsRenamePolicy parse_feilds_rename_policy(const ClassData &json_encoder) noexcept {
```

Comment:
AFAIK, we usually avoid using ClassData directly and use ClassPtr as a smart pointer to ClassData.
So instead of dereferencing a ClassPtr and passing it here, maybe accept `ClassPtr` in this method instead?
Same in other places.

Reply:
Yea, I know. But from my POV it's better to dereference ClassPtr in place where we know it's not empty and pass real ClassData object.